usr/lib/common/mech_aes.c

```diff
@@ -1191,6 +1292,189 @@ CK_RV aes_ctr_decrypt_update(STDLL_TokData_t *tokdata,
     }
 }
 
+//
+//
+CK_RV aes_xts_encrypt_update(STDLL_TokData_t *tokdata,
```

Comment:
Regarding multi-part support:
I don't think that it will work that way.
XTS is somewhat special when it comes to chaining. 
See https://en.wikipedia.org/wiki/Disk_encryption_theory#/media/File:XTS_mode_encryption.svg
So the last 2 blocks need some special handling I guess. All parts before are easy for chaining.
I guess you need to buffer up to 2 blocks of input data in an AES-XTS context structure.